src/extensions/renderer-api/catalog.ts

```diff
@@ -48,6 +48,33 @@ export class CatalogEntityRegistry {
   getItemsForCategory<T extends CatalogEntity>(category: CatalogCategory): T[] {
     return registry.getItemsForCategory(category);
   }
+
+  /**
+   * Add a onRun hook to a catalog entity.
+   * @param catalogEntityUid The uid of the catalog entity
+   * @param onRunHook The function that should return a boolean if the onRun of catalog entity should be triggered.
+   * @returns A function to remove that hook
+   */
+  addOnRunHook(catalogEntityUid: CatalogEntity["metadata"]["uid"], onRunHook: CatelogEntityOnRunHook) {
+    return registry.addOnRunHook(catalogEntityUid, onRunHook);
+  }
+
+  /**
+   * Returns one catalog entity onRun hook by catalog entity uid
+   */
+  getOnRunHook(catalogEntityUid: CatalogEntity["metadata"]["uid"]): CatelogEntityOnRunHook | undefined {
+    return registry.getOnRunHook(catalogEntityUid);
+  }
+
```

Comment:
We should support several onRunHooks. You sort of do already but not very well since you only ever return the first hook.

Reply:
I think this should be another PR,

Reply:
No because this will have to be supported until the next major

Reply:
At least in current timeframe I don't think I can do this before 5.3. @Nokel81 would you like to to take over on the multiple hooks part?